mvcc/watcher_group.go

```diff
@@ -232,14 +234,25 @@ func (wg *watcherGroup) choose(maxWatchers int, curRev, compactRev int64) (*watc
 		maxWatchers--
 		ret.add(w)
 	}
-	return &ret, ret.chooseAll(curRev, compactRev)
+	return &ret, ret.chooseAll(lg, curRev, compactRev)
 }
 
-func (wg *watcherGroup) chooseAll(curRev, compactRev int64) int64 {
+func (wg *watcherGroup) chooseAll(lg *zap.Logger, curRev, compactRev int64) int64 {
 	minRev := int64(math.MaxInt64)
 	for w := range wg.watchers {
 		if w.minRev > curRev {
-			panic(fmt.Errorf("watcher minimum revision %d should not exceed current revision %d", w.minRev, curRev))
+			if !w.restore {
+				panic(fmt.Errorf("watcher minimum revision %d should not exceed current revision %d", w.minRev, curRev))
+			}
+			if lg != nil {
```

Comment:
probably just remove the logging and add some comments on why this can happen on the restore case.

Reply:
Sure!